docs: Flex log files description and download

jwwojak opened the pull request:
# Overview

Having more info about Flex log files in the manual is helpful. Also, and perhaps more important, having this in the Flex manual allows us to get rid of another bad Help Center article ([How to download the logs on Opentrons Flex](https://support.opentrons.com/s/article/How-to-download-the-logs-on-Opentrons-Flex)).

For this article, we'll repurpose the intro and file download instructions from the Flex service manual, but without all the extra details used by the service techs.

See also RTC-824

Sandbox docs:
- Top-level section re-org: https://sandbox.docs.opentrons.com/docs-download-log-files/flex/advanced-operation/
- Log files: https://sandbox.docs.opentrons.com/docs-download-log-files/flex/advanced-operation/log-files/

## Test Plan and Hands on Testing

- Work through the instructions to make sure the steps are correct.
- Check to ensure images render.

## Changelog

On the surface, a simple change or PR: add steps on log file download.

I'd recommend some reorganization. This belongs with the other "advanced" operations articles. I'd like to:

- Break up the content in `advanced-operations.md`
- Make this section a multi-doc section to now include:
    - An intro: `advanced-operations/index.md` to summarize the sections.
    - Make the command line / SSH piece a separate file.
    - Make the Jupyter Notebook a separate file
    - Add this log file as a separate file

## Review requests

Consider the reorg. Is there another place to put this _without_ changing the structure?

## Risk assessment

Low, docs changes only.

jwwojak committed:
Reorganize sections and add new text

Read the description in the PR.

jwwojak committed:
Merge branch 'edge' into docs-download-log-files

jwwojak committed:
Malformed anchor link

jwwojak committed:
Fixing anchors and cross-reference links

Proposed changes will break some links between pages. Checked mkdocs output for warnings and fixed broken links.

Replaced "reference" style links with typical [ ]( ) style relative links.

jwwojak committed:
Intro and body revisions

There are 2 new log files: can_bus.log and update_server.log. I think I know what the former is, but not the update_server. checking on these to verify.

jwwojak committed:
Add summary text and final tweaks

jwwojak committed:
Fix sequential numbers in each step

ecormany commented on docs/flex/docs/advanced-operation/log-files.md:
```diff
@@ -0,0 +1,38 @@
+
+In an idle state or during a protocol run, your Flex constantly writes data to several different log files. Each of these files tracks the activities specific to different parts of the robot and its attachments. The following table summarizes the data captured in each file.
+
+| Log file | Description |
```
Make the first column wider so the file names fit without wrapping.

jwwojak replied:
I tried to be clever and apply a CSS rule to prevent line splitting on the first column:

> .first-col-no-wrap th:first-child,
.first-col-no-wrap td:first-child {
  white-space: nowrap;
}

This didn't work. Implemented an ugly-er fix. Might continue to test this separately.

jwwojak replied:
My other fix didn't work. Trying this instead. I think it is good enough.

ecormany commented on docs/flex/docs/advanced-operation/log-files.md:
```diff
@@ -0,0 +1,38 @@
+    !!!note
+        Flex produces a single `.zip` file that contains the logs. The compressed file name includes the robot’s name followed by `_logs.zip`. For example, if your robot is called “Flex1," the log file will be named `Flex1_logs.zip`.
+
+5. _(Optional)_ To read the logs, double-click the downloaded file to decompress it. The individual logs will expand into a new folder in the same location as the downloaded file. Any simple text editor should be able to open these files.
```
maybe remove "simple"?

jwwojak replied:
Removing

ecormany commented on docs/flex/docs/advanced-operation/command-line.md:
```diff
@@ -1,43 +1,44 @@
+Opentrons Flex gives you command-line access to its operating system through a Secure Shell (SSH) terminal connection. Terminal access lets you:
 
-## Command line operation over SSH
+- Run protocols directly via the [Python API and command line](https://docs.opentrons.com/v2/new_advanced_running.html#command-line).
```
New URL here: https://docs.opentrons.com/v2/advanced_control/command_line.html#command-line

jwwojak replied:
Updating URL.

ecormany commented on docs/flex/docs/advanced-operation/command-line.md:
```diff
@@ -1,43 +1,44 @@
 !!!note
-    - SSH keys are required before you can connect to Flex and issue commands from a terminal.
-    - If you're unable to use a Wi-Fi network for SSH, see [Hardwired SSH Connections][hardwired-ssh-connections] below.
+    If you're unable to use a Wi-Fi network for SSH, see [Making a hardwired SSH connection][making-a-hardwired-ssh-connection] below.
```
Should this note go in the "Making a wireless SSH connection" section instead?

jwwojak replied:
Yes. That's odd. It should be in that section. Moving.

ecormany reviewed (commented):
The new text is good. I agree that we should discuss the overall structure, specifically whether we think _downloading_ the logs qualifies as an "advanced operation".

ecormany commented on docs/flex/docs/advanced-operation/log-files.md:
```diff
@@ -0,0 +1,38 @@
+---
+title: "Opentrons Flex: Flex Log Files"
```
```suggestion
title: "Opentrons Flex: Log Files"
```

ecormany committed:
simplify log files page title

jwwojak committed:
Changes based on review comments

- update links
- re-org TOC, put log files first.
- fix table spacing
- found errant trailing whitespace and missing closing } in CSS file, when working with a table rule

jwwojak committed:
Log file table changes

Trying to avoid line wrapping in the first col of the log file table. I do

ecormany reviewed (approved):
Looks good!

jwwojak commented:
Squashed and merged. As this is Friday, it's probably better to wait for Monday to deploy.
